websites/G/GenoAnime/presence.ts

```diff
@@ -0,0 +1,119 @@
+}),
+strings = presence.getStrings({
+  play: "presence.playback.playing",
+  pause: "presence.playback.paused"
+}),
+browsingStamp = Math.floor(Date.now() / 1000);
+     
+let timestamps: number[],
+  video:HTMLVideoElement,
+  currentTime: number,
+  duration: number,
+  paused: boolean,
+  iFrameVideo: boolean,
+  playback:boolean,
+  synopsis_link: string; 
+
+  presence.on(
+    "iFrameData",
+    (data: {
+      iframe_video: {
+        duration: number;
+        iFrameVideo: boolean;
+        currTime: number;
+        paused: boolean;
+      };
+    }) => {
+      playback = data.iframe_video.duration !== null ? true : false;
+      if (playback) {
+        iFrameVideo = data.iframe_video.iFrameVideo;
+        currentTime = data.iframe_video.currTime;
+        duration = data.iframe_video.duration;
+        paused = data.iframe_video.paused;
+      }
+    }
+  );
+  
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "geno",
+    startTimestamp: browsingStamp
+
+  },
+
+title = document.title.slice(0, -13); //title of the page
+  
+  if (document.location.pathname == "/" ) {
+    presenceData.details = "Exploring Genoanime";
+  } else if (document.location.pathname.includes("/browse")) {
+    presenceData.details = "Exploring Genoanime library";
+    presenceData.buttons = [{label:"View Library",url: document.location.href}];
+  }
+  else if (document.location.pathname.includes("/details")) {
+    presenceData.details = "Browsing...";
+  } 
+  else if (document.location.pathname.includes("/watch")) {
+    presenceData.details = title;
+    presenceData.state = 'Episode '+ String(document.location.href.split("episode=")[1]);
+    synopsis_link = String(document.querySelector("#anime_details_breadcrumbs"));
+    presenceData.buttons = [{label:"Watch Episode",url: document.location.href},{label:"Check Synopsis",url: String(synopsis_link)}];
+
+    if (iFrameVideo) {
+      timestamps = presence.getTimestamps(
+        Math.floor(currentTime),
+        Math.floor(duration)
+      );
+    } else{
+      video = document.querySelector(
+        "div > div.plyr__video-wrapper > video"
+      );
+      if (video){
+        (currentTime = video.currentTime),
+        (duration = video.duration),
+        (paused = video.paused),
+        (timestamps = presence.getTimestamps(
+          Math.floor(currentTime),
+          Math.floor(duration)
+        ));
+      }
+    }
+    
+    if (!isNaN(duration)) {
+      presenceData.smallImageKey = paused ? "pause" : "play";
+      presenceData.smallImageText = paused
+        ? (await strings).pause
+        : (await strings).play;
+      presenceData.startTimestamp = timestamps[0];
+      presenceData.endTimestamp = timestamps[1];
+      if (paused) {
+        delete presenceData.startTimestamp;
+        delete presenceData.endTimestamp;
+      }
+    }
+    else if (isNaN(duration)){presenceData.startTimestamp = browsingStamp}
```

Comment:
For what purpose this else if exists if you already assign startTimestamp in presenceData object

Reply:
removed unnecessary statement